frontend/src/WorkflowEditor/FlowBuilder/index.jsx

```diff
@@ -49,54 +52,66 @@ function FlowBuilder(_props) {
   // Here we add a new node
   const onConnectEnd = useCallback(
     (event) => {
+      const { top, left } = flowElement.current.getBoundingClientRect();
+      const x = event.clientX - left - 75;
+      const y = event.clientY - top;
       const startingNodeId = editingActivity.nodeId;
       const startingNodeHandleId = editingActivity.handleId;
-
+      setShowBlockOptions({
+        x,
+        y,
+        startingNodeId,
+        startingNodeHandleId,
+        clientX: event.clientX,
+        clientY: event.clientY,
+      });
       setEditingActivity({ type: 'IDLE' });
 
       if (event.target.className === 'react-flow__pane') {
```

Comment:
should we remove this condition, I am not using it.

Reply:
The idea of the condition is to not create a new node if the dragging end point is another node or node handle. `react-flow__pane` represents the empty pane.

Reply:
Coolio. I will move the condition back into it. :)